packages/custom_lint/test/cli_process_test.dart

```diff
@@ -8,205 +8,188 @@ import 'cli_test.dart';
-    });
-
-    test('found lints', () async {
-      final plugin = createPlugin(name: 'test_lint', main: oyPluginSource);
-
-      final app = createLintUsage(
-        name: 'test_app',
-        source: {
-          'lib/main.dart': 'void fn() {}',
-          'lib/another.dart': 'void fail() {}',
-        },
-        plugins: {'test_lint': plugin.uri},
-        createDependencyOverrides: true,
-      );
-
-      final process = await TestProcess.start(
-        'dart',
-        [
-          'run',
-          'custom_lint',
-          '.',
-        ],
-        workingDirectory: app.path,
-      );
-
-      expect(process.stderr, emitsThrough(emitsDone));
-      expect(
-        process.stdout,
-        emitsInOrder(
-          [
-            '  lib/another.dart:1:6 • Oy • oy',
-            '  lib/main.dart:1:6 • Oy • oy',
-            emitsDone,
-          ],
-        ),
-      );
-      expect(process.exitCode, completion(1));
-    });
-
-    test('missing package_config.json', () async {
-      final plugin = createPlugin(name: 'test_lint', main: oyPluginSource);
-
-      final app = createLintUsage(
-        name: 'test_app',
-        source: {
-          'lib/main.dart': 'void fn() {}',
-          'lib/another.dart': 'void fail() {}',
-        },
-        plugins: {'test_lint': plugin.uri},
-        createDependencyOverrides: true,
-      );
-
-      // Create a child context root
-      final innerContextRoot = createLintUsage(
-        name: 'test_project_inner',
-        source: {
-          'lib/main.dart': 'void fn() {}',
-          'lib/another.dart': 'void fail() {}',
-        },
-        parent: app,
-        createDependencyOverrides: true,
-      );
-
-      // create error during initialization because of missing package_config.json
-      final packageConfig = File(
-        p.join(innerContextRoot.path, '.dart_tool', 'package_config.json'),
-      );
-      // Potentially resolve the file system link, temp folders are links on macOs into /private/var
-      final missingPackageConfig = await packageConfig.resolveSymbolicLinks();
-      packageConfig.deleteSync();
-
-      final process = await TestProcess.start(
-        'dart',
-        [
-          'run',
-          'custom_lint',
-          '.',
-        ],
-        workingDirectory: app.path,
-      );
-
-      expect(process.stdout, emitsDone);
-      expect(
-        process.stderr,
-        emitsThrough(
-          emitsInOrder(
-            [
+    test(
+      'no issues found',
+      timeout: timeout,
+      () async {
+        final plugin = createPlugin(name: 'test_lint', main: emptyPluginSource);
+
+        final app = createLintUsage(
+          name: 'test_app',
+          source: {
+            'lib/main.dart': 'void fn() {}',
+            'lib/another.dart': 'void fail() {}',
+          },
+          plugins: {'test_lint': plugin.uri},
+          createDependencyOverrides: true,
+        );
+
+        final process = await TestProcess.start(
```

Comment:
Honestly I'd remove TestProcess and stick to Process.run

Reply:
I don't know, then we'd have to remove even the `StreamMatcher`s for the error cases because `Process` returns binary streams. I can do whatever, but I think the 2 error cases are worth keeping atm.

Reply:
You can use `utf8.decode` to convert the binary

Reply:
Yea sure, but that doesn't really change anything - I don't think the dependency is a problem.

Reply:
Well I don't like this dependency being here.